cluster_experiments/washover.py

```diff
@@ -242,5 +242,124 @@ def from_config(cls, config) -> "Washover":
         return cls(washover_time_delta=washover_time_delta)
 
 
+class TwoEventsWashover(Washover):
+    """
+    For each record in our experiment, two events' timestamps,
+    login_timestamp and logout_timestamp.
+     We want to apply washover such that if there is a change in treatment
+    between login_timestamp and logout_timestamp.
+    Ex:
+    Calendar:
+        +-----------+-------+
+        | treatment | time  |
+        +-----------+-------+
+        | A         | 10:00 |
+        | B         | 11:00 |
+        | B         | 12:00 |
+        +-----------+-------+
+    Record data:
+
+        +----+------------+----------+
+        | id | start_time | end_time |
+        +----+------------+----------+
+        |  1 | 10:50      | 10:59    |
+        |  2 | 10:51      | 11:01    |
+        |  3 | 11:01      | 11:05    |
+        |  4 | 11:01      | 12:01    |
+        |  5 | 12:01      | 12:05    |
+        +----+------------+----------+
+
+    This washover will drop record number 2.
+    """
+
+    def __init__(
+        self,
+        calendar_df: pd.DataFrame,
```

Comment:
I'd try to avoid do to init with a calendar df, we've tried to keep constructors only based on very simple types like str, int and float. I'd assume the user already has the calendar in the record df, so you don't need to do a join, and it works like other washover classes

Reply:
okis